### PR TITLE
enhanced the extension

### PR DESCRIPTION
**Pull request by thekavikumar**
Issue: #7 

1. **Visual Feedback in Editor**: Matching columns in the CSV file are now highlighted directly within the Visual Studio Code editor. This provides users with immediate visual feedback on which columns match the columns defined in the selected Prisma schema.

2. **Interactive Interface for Prisma Schema Selection**: Users can now select the Prisma schema file using Visual Studio Code's built-in file picker dialog. This simplifies the process of specifying the Prisma schema file path, enhancing the user experience.

3. **Informative Messages**: When matching columns are found, the extension displays informative messages to indicate which model the columns match. This helps users quickly understand the relationship between the CSV columns and the Prisma schema models.

These are the features added.

### CODE REVIEW

**Comment by thekavikumar** [comment]
https://github.com/SamagraX-Stencil/stencil/assets/50093149/5c31d7b8-953f-4e70-93cf-1054e4307343

**Comment by thekavikumar** [comment]
@techsavvyash please review the PR

File: vsc-extension/.gitignore
Comment (techsavvyash) [inline vsc-extension/.gitignore:1]: Why have we added prisma in the gitignore?

Comment (thekavikumar) [reply vsc-extension/.gitignore:1]: I was testing with prisma folder

Comment (thekavikumar) [reply vsc-extension/.gitignore:1]: shall i remove the gitignore?

**Comment by techsavvyash** [comment]
> vscode.mp4

Why do we have to select the schema.prisma manually?

**Comment by thekavikumar** [comment]
> > vscode.mp4
> 
> Why do we have to select the schema.prisma manually?

because if we declare it in the extension file itself, and we deploy the extension the user can't modify the path as because he doesn't have the source code of the extension.

**Comment by techsavvyash** [comment]
What new have you done in this? This functionality is already present in the current state of the VS Code extension, seems more like a downgrade where we are picking the schema.prisma by hand and we have to right click after selecting data to validate and not just run it like a command from the VSCode command pallate.

**Comment by thekavikumar** [comment]
> What new have you done in this? This functionality is already present in the current state of the VS Code extension, seems more like a downgrade where we are picking the schema.prisma by hand and we have to right click after selecting data to validate and not just run it like a command from the VSCode command pallate.

no, we can use the command pallate to

**Comment by thekavikumar** [comment]
sir when we deploy the extension, how the user can access the source code? It is not possible right, for that only I've added option to select the file.

**Comment by techsavvyash** [comment]
> > > vscode.mp4
> > 
> > 
> > Why do we have to select the schema.prisma manually?
> 
> because if we declare it in the extension file itself, and we deploy the extension the user can't modify the path as because he doesn't have the source code of the extension. @techsavvyash

We should provide an option to configure the path to schema file using the extension settings if the user absolutely needs it, but in majority cases the `schema.prisma` of concern would always be present in the root level `prisma` folder.

**Comment by techsavvyash** [comment]
> sir when we deploy the extension, how the user can access the source code? It is not possible right, for that only I've added option to select the file.

why hard code the path? use environment variables/config parameters with a default value.

**Comment by thekavikumar** [comment]
so, is the pr okay? or any other features should be implemented sir?

**Comment by techsavvyash** [comment]
no, the PR is not okay, we are discussing the problems only.

**Comment by thekavikumar** [comment]
sir, what is the solution you suggest for these problems?

**Comment by techsavvyash** [comment]
please read the conversation again.

**Comment by thekavikumar** [comment]
yeah sir, so we can fetch it from env

**Comment by thekavikumar** [comment]
also please suggest me new features that can be implemented, I'm running out of ideas

**Comment by techsavvyash** [comment]
i am still not seeing what is the new thing you have implemented till now, the current functionlaity you are giving is already present in the current state of the extension, what is the point of this PR?

**Comment by thekavikumar** [comment]
It is to make some enhancement to the existing extension sir

**Comment by techsavvyash** [comment]
What is the enhacement?

**Comment by thekavikumar** [comment]
highlighting the matched columns, and configure path to prisma file instead of hard coding it in the extension itself

**Comment by techsavvyash** [comment]
> highlighting the matched columns, and configure path to prisma file instead of hard coding it in the extension itself

is the second part implemented?

**Comment by thekavikumar** [comment]
which one? option to configure the path to prisma file

**Comment by thekavikumar** [comment]
or the highlighting?

**Comment by techsavvyash** [comment]
also, it's better to highligh the unmatched columns (errors) rather than the matched columns.

**Comment by techsavvyash** [comment]
> which one? option to configure the path to prisma file?

this one

**Comment by thekavikumar** [comment]
> > which one? option to configure the path to prisma file?
> 
> this one

that's the selecting prisma file from file explorer that pops up

**Comment by thekavikumar** [comment]
> also, it's better to highligh the unmatched columns (errors) rather than the matched columns.

alright will do that

**Comment by techsavvyash** [comment]
> > > which one? option to configure the path to prisma file?
> 
> > 
> 
> > this one
> 
> 
> 
> that's the selecting prisma file from file explorer that pops up

I don't think picking files from the file picker counts as giving the path via a config. Stencil is built with the idea to "enhance" DX not downgrade it.

**Comment by thekavikumar** [comment]
alright sir, will close this pr, and will implement some other feature. 
can you please suggest some new feature that i can work on?

**Pull request closed without merging**